src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/GestureManager.shared.cs

```diff
@@ -202,13 +204,52 @@ internal void OnTapped(TouchEffect sender)
 			if (!sender.CanExecute || (sender.LongPressCommand != null && sender.InteractionStatus == TouchInteractionStatus.Completed))
 				return;
 
+			if (Device.RuntimePlatform == Device.Android)
+				HandleCollectionViewSelection(sender);
+
 			if (sender.Element is IButtonController button)
 				button.SendClicked();
 
 			sender.Command?.Execute(sender.CommandParameter);
 			sender.RaiseCompleted();
 		}
 
+		void HandleCollectionViewSelection(TouchEffect sender)
+		{
+			Element element = sender.Element;
+			while (element.Parent != null)
+			{
+				if (!(element.Parent is CollectionView collectionView))
+				{
+					element = element.Parent;
+					continue;
+				}
```

Comment:
Can we move this to a Helper class, you can use this implementation if you want
https://github.com/xamarin/Xamarin.Forms/blob/80a4dbfb1cd897f478c38d0aafa6a5d67940d2f3/Xamarin.Forms.Platform.iOS/Extensions/ViewExtensions.cs

Reply:
Those implementations are **internal** :(
But yes, I agree to move this logic :)

Reply:
Also, my implementation searches for an element where the parent type is TYPE (instead of the parent with type)

Reply:
Added two new helper methods